### PR TITLE
mmake: init at 1.2.0

### PR DESCRIPTION
**Pull request by gabesoft**
###### Motivation for this change
Added the [mmake](https://github.com/tj/mmake) package

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
This should be build from source.

**Comment by gabesoft** [comment]
@FRidh I was trying to build from source but I didn't get it to work. I guess I'm kind of unfamiliar with setting up the build step for `go` packages. Do you have any suggestions on how I can do this (an example package would help)

**Comment by gabesoft** [comment]
@FRidh I finally figured it out. I've update the package to build from sources.

File: pkgs/tools/misc/mmake/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    sha256 = "1pyqgk04v0f7a28cwq9c40bd2cgrkrv4wqcijdzpgn4bqhrqab4f";
+  };
+
+  goDeps = ./deps.nix;
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/tj/mmake;
+    description = "Mmake is a small program which wraps make to provide additional functionality, such as user-friendly help output, remote includes, and eventually more. It otherwise acts as a pass-through to standard make.";
+    license = licenses.mit;
+    platforms = platforms.all;
```
Comment (Mic92) [inline pkgs/tools/misc/mmake/default.nix]: Do you want to maintain this package?

Comment (gabesoft) [reply pkgs/tools/misc/mmake/default.nix]: Sure.  I added myself to maintainers.

**Pull request merged**